### PR TITLE
BIP1: Clarify and consolidate text regarding early stage idea championing

### PR DESCRIPTION
**Pull request by kanzure**
The BIP1 text has been updated to clarify how early stage idea championing works.

The text itself is not significantly different, for the most part these changes involved reordering the existing sentences and paragraphs, and removing some conflicting ambiguous details. LIkewise, the previous reasons for BIP editor rejection of a (e.g. malformed) BIP have been consolidated into closer proximity in the text. I suspect that this will help readers understand the process better.

Also, this branch includes some trivial boring BIP1 changes like:
* "Bitcoin mailing list" -> "bitcoin-dev mailing list"
* "Bitcoin repository" -> "relevant Bitcoin repository" (originally I was going for "Bitcoin Core" but Luke-Jr mentioned a good NACK reason)
* "Bitcoin issue tracker" -> "relevant Bitcoin issue tracker" (same reasoning)

I originally brought up the issue (of BIP1 early-stage championing ambiguity) in the bitcoin-dev IRC channel on freenode a few hours ago:
http://bitcoinstats.com/irc/bitcoin-dev/logs/2016/01/01#l1451664846.0

Only somewhat related, but I found this email interesting:
https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-September/010955.html

This pull request text was written as of commit 66224130f03ffb94daae9692d1b8d25b1c35e6ba.

### CODE REVIEW

File: bip-0001.mediawiki [hunk 1]
```diff
@@ -23,30 +24,26 @@ There are three kinds of BIP:
...
 
-Vetting an idea publicly before going as far as writing a BIP is meant to save the potential author time. Many ideas have been brought forward for changing Bitcoin that have been rejected for various reasons. Asking the Bitcoin community first if an idea is original helps prevent too much time being spent on something that is guaranteed to be rejected based on prior discussions (searching the internet does not always do the trick). It also helps to make sure the idea is applicable to the entire community and not just the author. Just because an idea sounds good to the author does not mean it will work for most people in most areas where Bitcoin is used.
+It is highly recommended that a single BIP contain a single key proposal or new idea. The more focused the BIP, the more successful it tends to be. Small enhancements or patches often don't need a BIP and can be injected into the Bitcoin Core development work flow with a patch submission to the Bitcoin Core issue tracker. If in doubt, split your BIP into several well-focused ones.
```
Comment (kanzure) [inline bip-0001.mediawiki]: This sentence is said twice:
> Small enhancements or patches often don't need a BIP and can be injected into the Bitcoin Core development work flow with a patch submission to the Bitcoin Core issue tracker.

wasn't intentional, should be removed/fixed.

**Comment by btcdrak** [comment]
I'd like to [quote @laanjw](https://github.com/bitcoin/bips/pull/195#issuecomment-138601045) for why BIP1 should be updated rather than creating a new BIP.

> I agree with that in general, but for the meta-bip I think it makes practical sense to make an exception there:
>
> People will go to BIP1 to learn about writing BIPs. Having it self-contained is a plus
> There are no implementations of BIP1 - unless you are sneaky and count the other BIPs, but that'd only matter if there are big structural changes which would 'invalidate' the other BIPs by BIP1

**Comment by evoskuil** [comment]
NACK: process references to bitcoin core and #bitcoin-dev

**Comment by kanzure** [comment]
@evoskuil These changes do not include a referene to the bitcoin-dev IRC channel on freenode. Also, is the one remaining use of "Bitcoin Core" at the end regarding committers still objectionable to you? It's pretty harmless, I think.

It was previously:

> Many BIPs are written and maintained by developers with write access to the Bitcoin codebase.

And in this changeset it is currently:

> Many BIPs are written and maintained by developers with write access to the Bitcoin Core codebase.

**Comment by evoskuil** [comment]
Hi @kanzure,

I don't see any reason for the reference to Bitcoin Core. One could just as easily change it to libbitcoin, given that several BIPs have been written by libbitcoin contributors as well (including this BIP1). But I would just drop this statement altogether. My mistake on the IRC, read too quickly.

**Comment by btcdrak** [comment]
@evoskuil I agree, drop the sentence entirely.

**Comment by kanzure** [comment]
@evoskuil @btcdrak removed in 66224130f03ffb94daae9692d1b8d25b1c35e6ba

**Comment by evoskuil** [comment]
@kanzure Change my NACK to ACK.

File: bip-0001.mediawiki [hunk 2]
```diff
@@ -23,30 +24,26 @@ There are three kinds of BIP:
...
 
-Authors MUST NOT self assign numbers, but should use an alias such as "bip-johndoe-infinitebitcoins" which includes the author's name/nick and the BIP subject.
+Vetting an idea publicly before going as far as writing a BIP is meant to both save the potential author and the wider community time. Many ideas have been brought forward for changing Bitcoin that have been rejected for various reasons. Asking the Bitcoin community first if an idea is original helps prevent too much time being spent on something that is guaranteed to be rejected based on prior discussions (searching the internet does not always do the trick). It also helps to make sure the idea is applicable to the entire community and not just the author. Just because an idea sounds good to the author does not mean it will work for most people in most areas where Bitcoin is used. Small enhancements or patches often don't need a BIP and can be injected into the relevant Bitcoin development work flow with a patch submission to the relevant Bitcoin issue tracker.
```
Comment (luke-jr) [inline bip-0001.mediawiki]: > to both save

Should be "to save both"

**Pull request merged**